operations/debug_notifications/get_bundle_diff.py

```diff
@@ -0,0 +1,199 @@
+    results = response.json()['results']
+    total_hits = response.json()['total_hits']
+    bundles = [format_bundle(r['bundle_fqid']) for r in results]
+
+    # The 'link' header refers to the next page of results to fetch. If there is no link header present,
+    # all results have been fetched.
+    # Example:
+    # link: <https://dss.dev.data.humancellatlas.org/v1/search?output_format=summary&replica=gcs&per_page=500&scroll_id=123>; rel="next"
+    link_header = response.headers.get('link', None)
+    while link_header:
+        next_link = link_header.split(';')[0]
+        next_url = next_link.strip('<').strip('>')
+        response = requests.post(next_url, json=query_json, headers=headers)
+        results = response.json()['results']
+        bundles.extend(format_bundle(r['bundle_fqid']) for r in results)
+        link_header = response.headers.get('link', None)
+    return bundles
+
+
+def query_dss_by_project_uuid(project_uuid: str, deployment: str) -> list:
+    """Query DSS to get the bundle list by project_uuid."""
+    query = deepcopy(DSS_PRIMARY_BUNDLES_QUERY)
+    query['es_query']['query']['bool']['must'][0]['match'] = {
+        "files.project_json.provenance.document_id": project_uuid
+    }
+    dss_url = (
+        'https://dss.data.humancellatlas.org'
+        if deployment == 'prod'
+        else f'https://dss.{deployment}.data.humancellatlas.org'
+    )
+    bundles_in_dss = get_bundles(query_json=query, dss_url=dss_url)
+    uuids = list(set([b['bundle_uuid'] for b in bundles_in_dss]))
+    return uuids
+
+
+def run(deployment: str, project: str, service_account_key: str, output: str):
+    # authenticate with Cromwell using OAuth (service account JSON key file)
+    auth = CromwellAuth.harmonize_credentials(
+        service_account_key=service_account_key,
+        url='https://cromwell.caas-prod.broadinstitute.org',
+    )
+
+    # query for workflows by project_uuid
+    query_dict = {
+        'label': {
+            'project_uuid': project,
+            'caas-collection-name': f'lira-{"int" if deployment == "integration" else deployment}',
+        },
+        'additionalQueryResultFields': ['labels'],
+    }
+
+    target_workflows = get_workflows(query_dict=query_dict, auth=auth)
+    uuids_triggered_workflows = [
+        get_bundle_uuid_from_workflow_label(workflow=wf) for wf in target_workflows
+    ]
+
+    print(f"Found {len(uuids_triggered_workflows)} bundles from workflows.")
+
+    uuids_in_dss = query_dss_by_project_uuid(
+        project_uuid=project, deployment=deployment
+    )
+
+    print(f"Found {len(uuids_in_dss)} bundles in DSS.")
+
+    print(
+        f'Found {len(set(uuids_in_dss) - set(uuids_triggered_workflows))} bundles that in DSS but did not trigger workflows'
+    )
+
+    results = {
+        'bundles_in_dss': uuids_in_dss,
+        'bundles_triggered_workflows': uuids_triggered_workflows,
+        'diff': list(set(uuids_in_dss) - set(uuids_triggered_workflows)),
```

Comment:
I'd suggest saving `set(uuids_in_dss) - set(uuids_triggered_workflows)` to a variable (e.g. `bundles_diff`) and then reuse that above.

Reply:
Done!